>   /home/gha/actions-runner/_work/llvm-project/llvm-project/polly/lib/Analysis/ScopBuilder.cpp:2981:7: error: no matching function for call to 'isSafeToLoadUnconditionally'
   2981 |   if (isSafeToLoadUnconditionally(LI->getPointerOperand(), LI->getType(),
        |       ^~~~~~~~~~~~~~~~~~~~~~~~~~~
  /home/gha/actions-runner/_work/llvm-project/llvm-project/llvm/include/llvm/Analysis/Loads.h:78:15: note: candidate function not viable: requires 4 arguments, but 5 were provided
     78 | LLVM_ABI bool isSafeToLoadUnconditionally(Value *V, Align Alignment,
        |               ^                           ~~~~~~~~~~~~~~~~~~~~~~~~~~
     79 |                                           const APInt &Size,
        |                                           ~~~~~~~~~~~~~~~~~~
     80 |                                           const SimplifyQuery &SQ);
        |                                           ~~~~~~~~~~~~~~~~~~~~~~~
  /home/gha/actions-runner/_work/llvm-project/llvm-project/llvm/include/llvm/Analysis/Loads.h:117:15: note: candidate function not viable: requires 4 arguments, but 5 were provided
    117 | LLVM_ABI bool isSafeToLoadUnconditionally(Value *V, Type *Ty, Align Alignment,
        |               ^                           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
    118 |                                           const SimplifyQuery &SQ);
        |                                           ~~~~~~~~~~~~~~~~~~~~~~~
  1 error generated.

--- a/llvm/include/llvm/Analysis/Loads.h
+++ b/llvm/include/llvm/Analysis/Loads.h
@@ -108,7 +108,7 @@ isReadOnlyLoop(Loop *L, ScalarEvolution *SE, DominatorTree *DT,
 
 /// Return true if we know that executing a load from this value cannot trap.
 ///
-/// If SQ.CxtI are specified this method performs context-sensitive analysis
+/// If SQ.CxtI is specified this method performs context-sensitive analysis
 /// and returns true if it is safe to load immediately before SQ.CxtI.
 ///
 /// If it is not obviously safe to load from the specified pointer, we do a

--- a/polly/lib/Analysis/ScopBuilder.cpp
+++ b/polly/lib/Analysis/ScopBuilder.cpp
@@ -2979,7 +2979,7 @@ isl::set ScopBuilder::getNonHoistableCtx(MemoryAccess *Access,
 
   auto &DL = scop->getFunction().getDataLayout();
   if (isSafeToLoadUnconditionally(LI->getPointerOperand(), LI->getType(),
-                                  LI->getAlign(), DL, nullptr)) {
+                                  LI->getAlign(), DL)) {
     SafeToLoad = isl::set::universe(AccessRelation.get_space().range());
   } else if (BB != LI->getParent()) {
     // Skip accesses in non-affine subregions as they might not be executed

--- a/polly/lib/Analysis/ScopDetection.cpp
+++ b/polly/lib/Analysis/ScopDetection.cpp
@@ -486,8 +486,7 @@ bool ScopDetection::onlyValidRequiredInvariantLoads(
 
     for (auto NonAffineRegion : Context.NonAffineSubRegionSet) {
       if (isSafeToLoadUnconditionally(Load->getPointerOperand(),
-                                      Load->getType(), Load->getAlign(), DL,
-                                      nullptr))
+                                      Load->getType(), Load->getAlign(), DL))
         continue;
 
       if (NonAffineRegion->contains(Load) &&

